### PR TITLE
Add support for guessing the model alias

### PR DESCRIPTION
**Pull request by ChrisMacNaughton**
When calling  functest-run-suite --keep-model --bundle $bundle,
Zaza will read the tests.yaml to determine the model alias to
use for a deployment, unless a model alias is explicitly passed
in with a colon. Additionally, if multiple bundles in the
tests.yaml match, Zaza will log that and fall back to the
default alias.

### CODE REVIEW

File: zaza/charm_lifecycle/func_test_runner.py
Comment (ajkavanagh) [inline zaza/charm_lifecycle/func_test_runner.py:204]: Why not initialise `matching_bundles` to a `set()`?

File: zaza/charm_lifecycle/func_test_runner.py
Comment (ajkavanagh) [inline zaza/charm_lifecycle/func_test_runner.py:210]: This would become `matching_bundles.add(alias)`.

File: zaza/charm_lifecycle/func_test_runner.py
Comment (ajkavanagh) [inline zaza/charm_lifecycle/func_test_runner.py:187]: Probably need some docs somewhere (here in the docstring?) to say that it does attempt to guess the bundle iff there is only one scoped alias?

**Review by ajkavanagh — COMMENTED** [review]
minor comments.  otherwise, looks great.

File: zaza/charm_lifecycle/func_test_runner.py
Comment (ChrisMacNaughton) [reply zaza/charm_lifecycle/func_test_runner.py:204]: Good point, I did look at that and ended up doing this beacuse I didn't pay enough attention to the `.opo()` method on a set :)

File: zaza/charm_lifecycle/func_test_runner.py
Comment (ChrisMacNaughton) [reply zaza/charm_lifecycle/func_test_runner.py:187]: will do, wanted to get another set of eyes on it early :)

**Review by ajkavanagh — APPROVED** [review]
LGTM - let me know if this is complete, and we can then merge it.  Nice addition to the library!

**Comment by ChrisMacNaughton** [comment]
I'm happy with it :)

**Pull request merged**
